pkg/types/azure/validation/disk.go

```diff
@@ -0,0 +1,45 @@
+package validation
+
+import (
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/openshift/installer/pkg/types/azure"
+)
+
+// ValidateDiskEncryption checks that the specified disk encryption configuration is valid.
+func ValidateDiskEncryption(p *azure.MachinePool, cloudName azure.CloudEnvironment, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	childFldPath := fldPath.Child("osDisk", "diskEncryptionSet")
+
+	diskEncryptionSet := p.OSDisk.DiskEncryptionSet
+	diskEncryptionSetID := diskEncryptionSet.ToID()
+	if diskEncryptionSetID != "" && cloudName == azure.StackCloud {
+		return append(allErrs, field.Invalid(childFldPath.Child("diskEncryptionSetID"), diskEncryptionSetID, "disk encryption sets are not supported not this platform"))
+	}
+	if !azure.RxSubscriptionID.MatchString(diskEncryptionSet.SubscriptionID) {
+		return append(allErrs, field.Invalid(childFldPath.Child("subscriptionID"), diskEncryptionSet.SubscriptionID, "invalid subscription ID format"))
+	}
+	if !azure.RxResourceGroup.MatchString(diskEncryptionSet.ResourceGroup) {
+		return append(allErrs, field.Invalid(childFldPath.Child("resourceGroup"), diskEncryptionSet.ResourceGroup, "invalid resource group format"))
+	}
+	if !azure.RxDiskEncryptionSetName.MatchString(diskEncryptionSet.Name) {
+		return append(allErrs, field.Invalid(childFldPath.Child("diskEncryptionSetName"), diskEncryptionSet.Name, "invalid name format"))
+	}
+	if !azure.RxDiskEncryptionSetID.MatchString(diskEncryptionSetID) {
+		return append(allErrs, field.Invalid(childFldPath.Child("diskEncryptionSetID"), diskEncryptionSetID, "invalid encryption set ID format"))
+	}
+
+	return allErrs
+}
+
+// ValidateEncryptionAtHost checks that the encryption at host configuration is valid.
+func ValidateEncryptionAtHost(p *azure.MachinePool, cloudName azure.CloudEnvironment, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	encryptionAtHost := p.EncryptionAtHost
+	if encryptionAtHost == true && cloudName == azure.StackCloud {
+		return append(allErrs, field.Invalid(fldPath.Child("encryptionAtHost"), encryptionAtHost, "encryption at host is not supported not this platform"))
```

Comment:
```suggestion
		return append(allErrs, field.Invalid(fldPath.Child("encryptionAtHost"), encryptionAtHost, "encryption at host is not supported on this platform"))
```
this pops up a few places if you can do find and replace

Reply:
Fixed.